content/swift/concepts/enums/enums.md

```diff
@@ -0,0 +1,151 @@
+---
+Title: 'Enums'
+Description: 'In Swift, an enumeration is a collection of set values that can be referenced throughout a program.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Enum'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+In Swift, an enumeration (enum) is a collection of set values that can be referenced throughout a program.
+
+## Syntax
+
+```swift
+enum MyEnum {
+  // Enumeration cases
+}
+```
+
+To access specific cases in `MyEnum`, the dot operator `.` is used, following the case name:
+
+```pseudo
+MyEnum.case
+```
+
+## Creating an `enum` and Examples
+
+`enum` cases can be defined on separate lines:
+
+```swift
+enum Level {
+  case beginner
+  case intermediate
+  case advanced
+}
+```
+
+`enum` cases can also be written in a single line for simplicity:
+
+```swift
+enum Level {
+  case beginner, intermediate, advanced
+}
+```
+
+### Raw Value
+
+Raw values can be added for each case if a raw type is defined after the `enum` name and trailing colon `:`.
+
+```swift
+enum Bishop : String {
+  case novice = "Beginner"
+  case firstJob = "Magician"
+  case secondJob = "Cleric"
+  case thirdJob = "Priest"
+}
+```
+
+The example above displays an `enum`, named `Bishop`, with a raw type of `String` and cases assigned to their respective raw values.
+
+### Associated Value
+
+Associated values can be added to available cases.
+
+```swift
+enum MageAdvancementTree {
+  case novice
+  case firstJob(String)
+  case secondJob(wizard: String)
+  case thirdJob(mage: String)
+  case fourthJob(arch: String)
+}
+
+var bishop = MageAdvancementTree.thirdJob(mage: "Priest")
+
+print(bishop)
+// Output: thirdJob(mage: "Priest")
+```
+
+An `enum` named `MageAdvancementTree` is created and most of its cases have associated values. Then, a `bishop` variable is declared as a `MageAdvancementTree` and sets the `mage` to a string, `"Priest"`.
+
+### `CaseIterable`
+
+By conforming to the [`CaseIterable`](https://developer.apple.com/documentation/swift/caseiterable) protocol, certain properties like `allCases` can be accessed:
```

Comment:
I noticed that Docs has [a Swift entry ](https://www.codecademy.com/resources/docs/swift/protocols) for protocols in general. Should we file an issue for someone to create a PR to either edit that entry or create a new one?

Reply:
I can go ahead and make that change. It's a built-in protocol so we can probably start out underneath an extra header. 

https://www.codecademy.com/resources/docs/swift/protocols
## Built-in Protocols

[`CaseIterable`](https://developer.apple.com/documentation/swift/caseiterable) has an `allCases` property. An enumeration can adopt this protocol to gain access to all its values.

Reply:
@Dusch4593 what do you think about this structure and maybe we can alternatively create a PR to add to this list of built-in protocols. In addition, this can be linked to this `enum` entry in the "Accessing an `enum` and Examples" section under the "Iterative" sub-section.

Reply:
Sounds great!